crates/cli-support/src/js/mod.rs

```diff
@@ -377,12 +377,13 @@ impl<'a> Context<'a> {
             // In `--target no-modules` mode we need to both expose a name on
             // the global object as well as generate our own custom start
             // function.
+            // `document.currentScript` property can be null in browser extensions
             OutputMode::NoModules { global } => {
                 js.push_str("const __exports = {};\n");
                 js.push_str("let script_src;\n");
                 js.push_str(
                     "\
-                    if (typeof document !== 'undefined') {
+                    if (typeof document !== 'undefined' && document.currentScript && document.currentScript.src) {
```

Comment:
```suggestion
                    if (typeof document !== 'undefined' && typeof document.currentScript !== 'null') {
```
Seems the spec says it can only be `HTMLScriptElement` or `null`.
https://html.spec.whatwg.org/multipage/dom.html#dom-document-currentscript-dev

Reply:
@daxpedda , changed as suggested, but I think `src` property can still be null. There is nothing in the spec about it being required and in at least one place it is implied as optional. We probably don't have to worry about those edge use cases.

Reply:
I guess we can deal with it if we actually **know** that it can happen, otherwise we would do checks like this literally everywhere.